AK/Optional.h

```diff
@@ -185,41 +218,41 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> {
     }
 
     template<typename... Parameters>
-    ALWAYS_INLINE void emplace(Parameters&&... parameters)
+    constexpr ALWAYS_INLINE void emplace(Parameters&&... parameters)
     {
         clear();
         m_has_value = true;
-        new (&m_storage) T(forward<Parameters>(parameters)...);
+        construct_at<RemoveConst<T>>(&m_storage, forward<Parameters>(parameters)...);
     }
 
     template<typename Callable>
-    ALWAYS_INLINE void lazy_emplace(Callable callable)
+    constexpr ALWAYS_INLINE void lazy_emplace(Callable callable)
     {
         clear();
         m_has_value = true;
-        new (&m_storage) T { callable() };
+        construct_at<RemoveConst<T>>(&m_storage, callable());
     }
 
-    [[nodiscard]] ALWAYS_INLINE bool has_value() const { return m_has_value; }
+    [[nodiscard]] constexpr ALWAYS_INLINE bool has_value() const { return m_has_value; }
 
-    [[nodiscard]] ALWAYS_INLINE T& value() &
+    [[nodiscard]] constexpr ALWAYS_INLINE T& value() &
     {
         VERIFY(m_has_value);
-        return *__builtin_launder(reinterpret_cast<T*>(&m_storage));
```

Comment:
I don’t quite understand what launder does, but I assume this was here for a reason to prevent erroneous devirtualization? Why is it safe to remove it?

Reply:
The code works in compile-time, so there is no UB in what's currently written. (My C++20 lifetime model also tells me that everything's fine with Leon's code)

Reply:
And in the first place, launder was there to prevent some GCC diagnostics 4 years ago: b9f30c6f2a69b0c07d9a61e5dd81f439a1eafc8f. Given that CI works with it now, there is no need to reintroduce it.

Reply:
Okay!